script.js

```
@@ -820,22 +816,56 @@ document.addEventListener('DOMContentLoaded', function() {
+                });
         }
     });
 
     // Helper function to update address from coordinates (mock for demo)
     function updateAddressFromCoordinates(lat, lng) {
-        // In a real app, you would use a geocoding service here
-        deliveryAddressInput.value = `${Math.round(lat*100)/100}, ${Math.round(lng*100)/100}`;
+        fetch(`https://nominatim.openstreetmap.org/reverse?format=jsonv2&lat=${lat}&lon=${lng}`)
+            .then(response => response.json())
+            .then(data => {
+                if (data && data.display_name) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)